pyang/statements.py

```diff
@@ -2118,6 +2118,47 @@ def v_reference_deviation(ctx, stmt):
     stmt.i_target_node = find_target_node(ctx, stmt)
 
 def v_reference_deviate(ctx, stmt):
+    def search_ancestor_config_false(node, target):
+        # the target node's config is set to 'true', and the ancestor node above it
+        # cannot have 'config' is 'false'
+        if node is None or node.parent is None:
+            return False
+        else:
+            p_config = node.parent.search_one('config')
+            if p_config is not None and p_config.arg == 'false':
+                err_add(ctx.errors, c.pos, 'INVALID_CONFIG', ())
+                return True
+            return search_ancestor_config_false(node.parent, target)
+
+    def search_children_config_true(node, target):
+        # the target node's config is set to 'false', and the node underneath it
+        # cannot have 'config' is 'true'
+        if node.keyword in data_definition_keywords:
+            c_config = node.search_one('config')
+            if c_config is not None and c_config.arg == 'true':
+                err_add(ctx.errors, c.pos, 'INVALID_CONFIG', ())
+                return True
+        else:
+            return False
+        if node is None or not hasattr(node, 'i_children'):
+            return False
+        else:
+            for children in node.i_children:
+                if search_children_config_true(children, target):
+                    return True
+            return False
+
+    def inherit_parent_i_config(node, value):
+        if node is None or not hasattr(node, 'i_children'):
+            return
+        else:
+            for children in node.i_children:
```

Comment:
Minor, but change "children" to "child".

Reply:
OK